mmeds/util.py

```diff
@@ -182,6 +182,58 @@ def join_metadata(subject, specimen, subject_type):
     return df
 
 
+def split_metadata(full, clean_for_meta_analysis=True, id_col=('RawData', 'RawDataID'), new_study_name=None):
+    """ Splits a full metadata file into two dataframes, subject and specimen """
+    # Determine subject column set and subject ID col
+    tables = [header[0] for header in full.columns]
+    if 'Ethnicity' in tables and 'Chow' in tables:
+        # Subject type mixed
+        subj_tables = fig.MIXED_SUBJECT_TABLES
+        subj_id_col = full[[
+            ('Subjects', 'HostSubjectId'),
+            ('AnimalSubjects', 'AnimalSubjectID')
+        ]].bfill(axis=1).iloc[:, 0]
+    elif 'Ethnicity' in tables:
+        # Subject type human
+        subj_tables = fig.SUBJECT_TABLES
+        subj_id_col = full[('Subjects', 'HostSubjectId')]
+    elif 'Chow' in tables:
+        # Subject type animal
+        subj_tables = fig.ANIMAL_SUBJECT_TABLES
+        subj_id_col = full[('AnimalSubjects', 'AnimalSubjectID')]
+    else:
+        raise ValueError("Unable to determine subject type from metadata file")
+    # Duplicate subject ID column
+    full[('AdditionalMetaData', 'SubjectIdCol')] = subj_id_col
+
+    # Get list of column headers based on config tables
+    subj_cols = [col for col in full.columns if col[0] in (subj_tables - {'AdditionalMetaData'})]
+    spec_cols = [col for col in full.columns if col[0] in fig.SPECIMEN_TABLES]
+
+    if set(spec_cols).intersection(set(subj_cols)):
+        raise ValueError("Cannot differentiate specimen and subject columns")
+
+    # Separate dfs by columns
+    subj_df = full[subj_cols]
+    spec_df = full[spec_cols]
+
+    if clean_for_meta_analysis:
```

Comment:
should almost be it's own method, what is the purpose of this? why might you want True or False

Reply:
added comments. for now it'll always be True, but it'll need to be False when we use it for generating sub-analyses